ethcore/builtin/src/lib.rs

```diff
@@ -213,93 +240,120 @@ impl Builtin {
 
 	/// Whether the builtin is activated at the given block number.
 	pub fn is_active(&self, at: u64) -> bool {
-		at >= self.activate_at
+		self.pricer.keys().rev().any(|&other_at| at >= other_at)
```

Comment:
`is_active()` is called from `machine` and `executive`; if this is a "hot" function maybe this onerous?

Reply:
you mean like this would be `n times` slower because we might need to iterate through the entire `BTreeMap` in the worst case instead of a constant time operation.

But, I think it is very likely that one or two iterations of the `BTreeMap`, otherwise is the transitions enabled too early I guess.

Still, accessing the first element through an iterator 20 times slower than the previous version on my machine :cry: but still if we want more than one activation at runtime it will be slower regardless....

So, I don't know

Reply:
I will investigate if it is easy to create a benchmark for this, in `machine`

Reply:
I benched it locally,

- Single activation, ~10% slower
- 10 activations where block number lower than lowest block number in the map: ~30% slower (i.e, most iterate through entire map `O(n)`)
- 10 activations block number bigger than highest block number, ~10% slower

Reply:
you could do a binary search with
`pricer.range(0..=at).last().is_some()`, not sure if it's faster for a small number of activations though

Reply:
It is definitely faster for 10 elements seems slower for 1 element though.

Let me check